### PR TITLE
Java Backend builtin map: fixed map.removeAll when attempting to remove symbolic elements

### PR DESCRIPTION
**Pull request by denis-bogdanas**
Also tests. Only tests remove-01-03 & 16... were passing before this fix.

Tests are based on tests for lookup.

### CODE REVIEW

File: k-distribution/tests/regression-new/map-symbolic-tests/remove [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//copied from lookup test plan
+
```
Comment (ehildenb) [inline k-distribution/tests/regression-new/map-symbolic-tests/remove]: Get rid of these two lines, I don't think the `gen-tests.sh` script will treat them correctly.

Also, you need to modify `gen-tests.sh` to also call `remove` in addition to `assignment lookup inkeys`.

**Review by ehildenb — CHANGES_REQUESTED** [review]

**Comment by ehildenb** [comment]
As far as checking that the tests give what you guys expect, I think @daejunpark should have a look as well, so I'll request him as a reviewer.

File: k-distribution/tests/regression-new/map-symbolic-tests/remove (hunk 1 above)
Comment (ehildenb) [reply k-distribution/tests/regression-new/map-symbolic-tests/remove]: The test files `remove-#-spec.k` should be populated by the `gen-tests.sh` script here.

Comment (denis-bogdanas) [reply k-distribution/tests/regression-new/map-symbolic-tests/remove]: I didn't use the script. What is it supposed to do?

Comment (denis-bogdanas) [reply k-distribution/tests/regression-new/map-symbolic-tests/remove]: Done.

**Comment by ehildenb** [comment]
The tests look good, if @daejunpark approves it should be ready for merge.

**Comment by denis-bogdanas** [comment]
@ehildenb I guess you should change your review status to "comment".

**Review by daejunpark — APPROVED** [review]

**Review by ehildenb — APPROVED** [review]

**Pull request merged**